src/theory/splitter.h

```diff
@@ -0,0 +1,138 @@
+/******************************************************************************
+ * Top contributors (to current version):
+ *   Amalee Wilson, Andrew Wu
+ *
+ * This file is part of the cvc5 project.
+ *
+ * Copyright (c) 2009-2021 by the authors listed in the file AUTHORS
+ * in the top-level source directory and their institutional affiliations.
+ * All rights reserved.  See the file COPYING in the top-level source
+ * directory for licensing information.
+ * ****************************************************************************
+ *
+ */
+
+#include "cvc5_private.h"
+
+#ifndef CVC5__THEORY__SPLITTER_H
+#define CVC5__THEORY__SPLITTER_H
+
+#include <math.h>
+
+#include <fstream>
+#include <list>
+#include <sstream>
+
+#include "options/smt_options.h"
+#include "proof/trust_node.h"
+#include "theory/valuation.h"
+
+namespace cvc5 {
+
+class TheoryEngine;
+
+namespace prop {
+class PropEngine;
+}
+
+namespace theory {
+
+class Splitter
+{
+ public:
+  Splitter(TheoryEngine* theoryEngine, prop::PropEngine* propEngine);
```

Comment:
It might make sense to make this class inherit from `EnvObj`, so that we can easily access, e.g., the options. (while you can still do `options::foo()`, this is deprecated and we should rather use `options().foo.bar`, given one has access to `options()` via the environment).
See for example `src/decision/decision_engine.(h|cpp)` for a simple class that is an `EnvObj`.